extensions/scheduler/deployment/src/main/java/io/quarkus/scheduler/deployment/SchedulerProcessor.java

```diff
@@ -117,27 +115,21 @@ public void transform(TransformationContext context) {
     }
 
     @BuildStep
-    void collectScheduledMethods(
-            SchedulerConfig config,
-            BeanArchiveIndexBuildItem beanArchives,
-            BeanRegistrationPhaseBuildItem beanRegistrationPhase,
-            BuildProducer<ScheduledBusinessMethodItem> scheduledBusinessMethods,
-            BuildProducer<BeanConfiguratorBuildItem> beans) {
+    void collectScheduledMethods(BeanArchiveIndexBuildItem beanArchives, BeanRegistrationPhaseBuildItem beanRegistrationPhase,
+            BuildProducer<ScheduledBusinessMethodItem> scheduledBusinessMethods) {
```

Comment:
You can't remove the `BuildProducer<BeanConfiguratorBuildItem> beans` even though it's unused. This ensures this build step is executed before `ArcProcessor#validate()`. See the `BeanRegistrationPhaseBuildItem` javadoc.

Reply:
Good catch. I might have gone a bit fast on that one. Let's keep it.

Reply:
Off topic: should we have a documentation-only annotation, called e.g. `@Ordering`, that would be added to such places to let people know what's the purpose of the dependency?

Reply:
Sounds like a very good idea.